### PR TITLE
CDK For Kubernetes (cdk8s) 2022 Annual Review

### PR DESCRIPTION
**Pull request by iliapolo**
Annual Review

### CODE REVIEW

File: reviews/2022-cdk8s-annual.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
+## Background
+
+CDK For Kubernetes (cdk8s) is an open-source software development framework for defining Kubernetes 
+applications and reusable abstractions. Users leverage familiar programming languages and rich object-oriented APIs to 
+create standard Kubernetes manifests which can be applied to any Kubernetes cluster.
+
+CDK8s joined the CNCF on November 10, 2022.
```
Comment (angellk) [inline reviews/2022-cdk8s-annual.md:7]: @iliapolo is this date accurate? [CNCF site](https://www.cncf.io/projects/cdk-for-kubernetes-cdk8s/) shows November 10, 2020

**Comment by angellk** [comment]
@iliapolo do you have a link to 2021 Annual Review please?

File: reviews/2022-cdk8s-annual.md (hunk 1 above)
Comment (iliapolo) [reply reviews/2022-cdk8s-annual.md:7]: Whoops, good catch! I'll change.

**Comment by iliapolo** [comment]
@angellk 

> @iliapolo do you have a link to 2021 Annual Review please?

We actually didn't do a 2021 review :\ - not sure how it got missed...

**Comment by angellk** [comment]
Thanks @iliapolo - no worries, I'll go from your application. If you have any other links from 2020, please share them with me!

File: reviews/2022-cdk8s-annual.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+The full devstats page for cdk8s is available here: https://cdk8s.devstats.cncf.io/d/8/dashboards
+
+* In the past year, there were 7 [community contributions](https://cdk8s.devstats.cncf.io/d/56/company-commits-table?orgId=1&from=now-1y&to=now&var-repogroups=All&var-companies=Palo%20Alto%20Networks&var-companies=Cisco&var-companies=GridLine) 
+from 3 different users, in 3 different organizations (Palo Alto Networks, GridLine, Cisco). There is an additional [contribution](https://github.com/cdk8s-team/cdk8s-cli/pull/324) 
+that doesn’t show up on devstats, probably because the user isn’t associated with any company.
+
+### How many maintainers do you have, and which organizations are they from?
+
+The project currently has 4 maintainers, all from AWS:
```
Comment (angellk) [inline reviews/2022-cdk8s-annual.md:23]: @iliapolo from https://github.com/cdk8s-team/cdk8s/blob/master/OWNERS.md it looks like there are 7 maintainers, all from AWS. Does OWNERS.md or this file need to be updated for accuracy?

File: reviews/2022-cdk8s-annual.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+2. Add support for Golang. Users have expressed desire to author cdk8s in applications in Golang, 
+it was in-fact our mostly highly voted [issue](https://github.com/cdk8s-team/cdk8s/issues/119) in the repository.
+3. Help customers safe guard their Kubernetes workloads by integrating cdk8s into policy compliant pipelines 
+and applying policy-as-code tools on the manifests generated by cdk8s. To that end, we created a spec and 
+a pluggable mechanism in the cdk8s-cli that allows third-party vendors to implement plugins for their 
+tools (e.g Kyverno, Checkov, Kubescape, ...). We have partnered with [Datree](https://www.datree.io/) to create a reference 
+implementation of one such plugin, available [here](https://github.com/datreeio/datree-cdk8s).
+
+Official announcements have been made via the AWS blog:
```
Comment (angellk) [inline reviews/2022-cdk8s-annual.md:63]: @iliapolo are there any vendor neutral blog sources you can share?

File: reviews/2022-cdk8s-annual.md (hunk 2 above)
Comment (iliapolo) [reply reviews/2022-cdk8s-annual.md:23]: One maintainer (Ryan Parker) needs to be removed as he is no longer involved in the project. Everything else is accurate.

File: reviews/2022-cdk8s-annual.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+We are currently aware of 7 companies who adopted cdk8s into their production workloads, 
+and an additional 9 who are currently evaluating it. We are in the process of asking them 
+to be added to the [ADOPTERS](https://github.com/cdk8s-team/cdk8s/blob/master/ADOPTERS.md) file.
+
+We are also seeing signs of increasing adoption by observing [NPM download stats](https://npm-stat.com/charts.html?package=cdk8s&from=2022-01-01&to=2023-02-23). 
+They show that over the last year, we have grown from ~12k to ~129k downloads a week for the core library. 
+GitHub stars have also increased from 2280 to [3492](https://star-history.com/#cdk8s-team/cdk8s&Date) stars in the course of the last 15 months. 
+
+In addition, the #cdk8s channel in the [cdk.dev](https://cdk.dev/) workspace now contains ~7k members and we
```
[the remark this answers is not in the document]
Comment (iliapolo) [reply reviews/2022-cdk8s-annual.md:40]: Added

File: reviews/2022-cdk8s-annual.md (hunk 3 above)
Comment (iliapolo) [reply reviews/2022-cdk8s-annual.md:65]: You mean blogs posted by maintainers in other sources? All our blogs currently use the AWS blog. There are many third-party blogs about cdk8s posted on other sources though - would you like links to those?

File: reviews/2022-cdk8s-annual.md (hunk 2 above)
Comment (iliapolo) [reply reviews/2022-cdk8s-annual.md]: Thanks for the PR! :)

Updated this file as well to reflect the correct state.

**Comment by angellk** [comment]
Copying comments here from the [TAG App Delivery issue for CDK8S annual review](https://github.com/cncf/tag-app-delivery/issues/432):

> In reviewing CDK8S, a few areas of concern:
> 
> Community meetings were suspended in March, 2022 and have not resumed
> The main slack channel ([cdk.dev](https://cdk.dev/)) for CDK8S is commingled with non-CNCF projects (the rest of CDK slack), the CNCF CDK8S slack channel has 7,000+ but is not active
> The CDK8S blog is hosted by AWS so does not appear vendor neutral
> There don't appear to be efforts to expand maintainers outside AWS (though project adoption itself seems healthy)

cc: @TheFoxAtWork @justincormack @kgamanji @AloisReitbauer @thschue @joshgav 

Recommendation is review by TOC for remediation or archival

**Comment by TheFoxAtWork** [comment]
@justincormack @kgamanji Would you both review and file a Health issue for the project? Its unclear if this needs to move to archival or roadmap for improvement given the information here. This project has been in Sandbox status for 3 years with little-to-no progress towards incubation (doesn't appear much has changed since acceptance).

**Comment by raravena80** [comment]
FWIW, I'd ping the cloudflare folks. I believe their k8s team was using this extensively to manage their clusters.

**Comment by amye** [comment]
Marking this as complete, annual reviews are no longer a requirement for sandbox. 
We'll be rolling out more efficient ways to show project health and metrics in 2024.

**Pull request merged**